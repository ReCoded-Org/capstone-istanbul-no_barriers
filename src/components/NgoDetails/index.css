.ngoDetails {
  width: 20rem;
}

td.ngoDetailsTableData {
  padding: 0;
  padding-right: 2rem;
}

.moreInfo {
  margin-right: 1rem;
}

.btnWebsite {
  margin-right: 2rem;
  background-color: transparent;
  border: 1px solid var(--dark-orange);
  color: var(--dark-orange);
}

.btnWebsite.btn-primary:not(:disabled):not(.disabled):active,
.btnWebsite:hover {
  background-color: var(--dark-orange);
  color: whitesmoke;
  border: none;
}

.btnAddFavourite,
.btnAddFavourite:hover,
.btnAddFavourite:focus {
  background-color: var(--orange);
  border: none;
}

/* filter favourite button active state style */
.btnAddFavourite.btn-primary:not(:disabled):not(.disabled):active {
  background-color: var(--dark-orange);
}