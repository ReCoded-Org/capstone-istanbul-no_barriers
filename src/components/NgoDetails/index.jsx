import React from "react";
import { Button } from "react-bootstrap";
import "./index.css";
import { useTranslation } from "react-i18next";

const NgoDetails = ({ details }) => {
  const { t } = useTranslation();

  return (
    <div className="ngoDetailsTableData">
      <h3>{details.facilityName}</h3>
      <p className="address">{details.address}</p>
      <h5>{t("resources.ngoList.ngoDetails.aidDescription")}</h5>
      <p className="ngoDetailsServiceDescription">
        {details.serviceDescription}
      </p>
      <p>
        {t("resources.ngoList.ngoDetails.responsiblePerson")}
        {details.responsiblePerson}
        <br />
        {t("resources.ngoList.ngoDetails.phoneNum")}
        {details.phoneNumber}
        <br />
        {t("resources.ngoList.ngoDetails.email")}
        {details.email}
      </p>
      <div className="d-flex align-items-baseline">
        <p className="moreInfo">{t("resources.ngoList.ngoDetails.moreInfo")}</p>
        <Button
          href={details.website}
          target="blank"
          className="btnWebsite shadow-none"
        >
          {t("resources.ngoList.ngoDetails.website")}
        </Button>
        <Button className="btnAddFavourite shadow-none">
          {t("resources.ngoList.ngoDetails.favourite")}
        </Button>
      </div>
    </div>
  );
};

export default NgoDetails;
