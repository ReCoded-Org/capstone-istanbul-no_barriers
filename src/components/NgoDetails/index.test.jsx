import React from "react";
import renderer from "react-test-renderer";
import NgoDetails from "./index";

it("NgoDetails renders correctly", () => {
  const details = {
    facilityName: "abc",
    address: "address",
    serviceDescription: "serviceDescription",
    responsiblePerson: "responsiblePerson",
    phoneNumber: 12344,
    email: "email",
    website: "website",
  };
  const tree = renderer.create(<NgoDetails details={details} />).toJSON();
  expect(tree).toMatchSnapshot();
});
