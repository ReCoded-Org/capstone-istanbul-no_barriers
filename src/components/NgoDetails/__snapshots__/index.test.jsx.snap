// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NgoDetails renders correctly 1`] = `
<div
  className="ngoDetailsTableData"
>
  <h3>
    abc
  </h3>
  <p
    className="address"
  >
    address
  </p>
  <h5>
    resources.ngoList.ngoDetails.aidDescription
  </h5>
  <p
    className="ngoDetailsServiceDescription"
  >
    serviceDescription
  </p>
  <p>
    resources.ngoList.ngoDetails.responsiblePerson
    responsiblePerson
    <br />
    resources.ngoList.ngoDetails.phoneNum
    12344
    <br />
    resources.ngoList.ngoDetails.email
    email
  </p>
  <div
    className="d-flex align-items-baseline"
  >
    <p
      className="moreInfo"
    >
      resources.ngoList.ngoDetails.moreInfo
    </p>
    <a
      className="btnWebsite shadow-none btn btn-primary"
      href="website"
      onClick={[Function]}
      onKeyDown={[Function]}
      target="blank"
    >
      resources.ngoList.ngoDetails.website
    </a>
    <button
      className="btnAddFavourite shadow-none btn btn-primary"
      disabled={false}
      type="button"
    >
      resources.ngoList.ngoDetails.favourite
    </button>
  </div>
</div>
`;
